core/src/smartcontracts/isi/asset.rs

```diff
@@ -54,89 +54,165 @@ pub mod isi {
         }
     }
 
-    impl<W: WorldTrait> Execute<W> for Mint<Asset, u32> {
-        type Error = Error;
+    /// Trait for asset values that could be converted to `f64` and observed by prometheus.
+    trait FloatMetricFrom {
```

Comment:
We need a separate trait because of the orphan rule, correct?

Reply:
Yeah, rust forbid to implement `From<u128>` for `f64`.